File: kotlin/countConstruct.kt
fun countConstruct(target: String, wordBank: Array<String>): Int {
    if (target.isEmpty()) return 1

    var count = 0;

    wordBank.forEach { word ->
        if (target.startsWith(word))
            count += countConstruct(target.substring(word.length), wordBank)
    }

    return count
}

fun main(): Unit {
    println(countConstruct("purple", arrayOf("purp", "p", "ur", "le", "purpl")))
}

File: javascript/countConstruct.js
function countConstruct(target, wordBank) {
    if (target.length === 0) return 1;

    let count = 0;

    for (let word of wordBank)
        if (target.startsWith(word))
            count += countConstruct(target.substr(word.length), wordBank);

    return count;
}

console.log(countConstruct("purple", ["purp", "p", "ur", "le", "purpl"]))